### ISSUE TITLE
Block helper don't render local dat objects when on a handlebars partial view

### ISSUE THREAD

**Issue by kovachjk** [wycats/handlebars-site#115]
I cannot get a block helper to render local data.
It works fine on html page and on development sites, but not as a partial.
I copied similar code from http://www.javascriptoo.com/handlebars-js.  This code runs on a html page but not when use on my favorite kittenView.hbs partial page in express-node.
```javascript
<div id="placeholder">This will get replaced by handlebars.js</div>
<h3>{{title}}</h3>
<script type="text/javascript" src="//cdnjs.cloudflare.com/ajax/libs/handlebars.js/1.0.rc.1/handlebars.min.js"></script>

<script id="myTemplate" type="text/x-handlebars-template">
    <ul>
        <li>Option 1</li>
        {{#each options}}
            <li {{hidden}}>{{name}}</li>
        {{/each}}
    </ul>
</script>
<script type="text/javascript">
    var source = document.getElementById("myTemplate").innerHTML;
    var template = Handlebars.compile(source);
    var results;
    var datax = {options:[
            {"hidden":"", "name": "foo" },
            {"hidden":"hidden",  "name": "bar" },
            {"hidden":"",  "name": "baz" }
        ]};
    results = template('{{data}}');
    console.log('data.name'+ '{{data.[1].name}}');
    document.getElementById("placeholder").innerHTML=results;
</script>
```

**Comment by kpdecker** [comment]
Can you create a fiddle showing this in action so we can try to debug this? The contributing doc has a link to a template that you can use as a basis.

**Comment by kovachjk** [comment]
Hi Kevin,
Thanks for the reply.

Unfortunately the only place it didn’t work was in express, where my siteView.hbs view held the offending code which was being consumed by the {{body}} portion of a template.

I am a relative newbie and in trying to determine what modules to use, I had both handlebars and express-handlebars node-modules in my project .  Maybe that was keeping it from working.  
I also tried calling handlebars in my view with
<script type="text/javascript" src="//cdnjs.cloudflare.com/ajax/libs/handlebars.js/1.0.rc.1/handlebars.min.js"></script>This also worked with just putting everything on a single html page and not using a template

I finally worked around it by building my list on the server and passing it with everything else.

I am a brut force programmer, I my gift is coming up with many different ways to solve a problem, which means I seldom remember how I solve things because once it’s solved I on to something else.  I now try to remember to stop and document solutions, but since I never solved it, I can’t remember everything I did.

Sorry I can’t be more help.
